src/simulate.jl

```diff
@@ -152,20 +149,17 @@ end
 function simulate!(
     rng::AbstractRNG, y::AbstractVector, m::LinearMixedModel{T}; β=m.β, σ=m.σ, θ=m.θ
 ) where {T}
-    length(β) == length(fixef(m)) ||
-        length(β) == length(coef(m)) ||
+    length(β) == length(m.feterm.piv) || length(β) == m.feterm.rank ||
```

Comment:
I don't think I understand this change. IIUC, the `piv` field of a model's `FeTerm` will always have length equal to `length(coef(model)) == size(modelmatrix(model), 2)`, and `length(fixef(model)) == rank(model)` in general but is less than `length(piv)` for rank deficiency. Assuming my understanding is correct, isn't this exactly the same as what was there?